### PR TITLE
Un-confine the mouse on game over

### PR DESCRIPTION
**Pull request by aaronfranke**
The mouse doesn't need to be confined on the game over screen, so un-confine it.

I also added a space to the project name and added a description.

### CODE REVIEW

File: player/player.gd [hunk 1]
```diff
@@ -129,6 +129,7 @@ func die():
 		return
 	dead = true
 
+	Input.set_mouse_mode(Input.MOUSE_MODE_VISIBLE)
```
Comment (akien-mga) [inline player/player.gd]: It would be cleaner to do this in `global.set_game_over()`.

File: level/ui.gd [hunk 2]
```diff
@@ -95,3 +95,4 @@ func show_score():
 
 func _on_restart_pressed():
 	global.restart()
+	Input.set_mouse_mode(Input.MOUSE_MODE_CONFINED)
```
Comment (akien-mga) [inline level/ui.gd]: This should be done in `global.restart()`.

Comment (akien-mga) [reply level/ui.gd]: Or maybe not, lemme think.

Comment (akien-mga) [reply level/ui.gd]: This should actually not be necessary, `restart()` reloads the scene and thus the mouse mode should be confined in `_ready`.

File: player/player.gd (hunk 1 above)
Comment (akien-mga) [reply player/player.gd]: Actually no, all mouse mode stuff is handled in `level/ui.gd` currently, so best move it there (so in `game_over()`).

File: level/ui.gd (hunk 2 above)
Comment (aaronfranke) [reply level/ui.gd]: I tested without it, the mouse doesn't confine itself unless I include this line.

**Comment by akien-mga** [comment]
On second thought I'd do something like this:
```diff
diff --git a/level/ui.gd b/level/ui.gd
index 88de755..627c5bd 100644
--- a/level/ui.gd
+++ b/level/ui.gd
@@ -21,11 +21,12 @@ const RANDOM_BITS = [
 
 func _ready():
 	$game_over.hide()
-	$info.hide()
 
 	if global.first_run:
 		global.first_run = false
 		show_info()
+	else:
+		hide_info()
 
 	$game_over/random_bit.set_bbcode(RANDOM_BITS[randi() % RANDOM_BITS.size()])
 
@@ -66,6 +67,7 @@ func _process(_delta):
 
 
 func game_over():
+	Input.set_mouse_mode(Input.MOUSE_MODE_VISIBLE)
 	$AnimationPlayer.play("game_over")
 	yield($AnimationPlayer, "animation_finished")
 	$AnimationPlayer.play("highlight_button")

```

**Comment by aaronfranke** [comment]
> @akien-mga: On second thought I'd do something like this:
> 
> ```diff
> +	else:
> +		hide_info()
> ```

For some reason this causes enemies to stop spawning after the first round. If I use `$info.hide()` in the `else:` it works.

**Comment by akien-mga** [comment]
> For some reason this causes enemies to stop spawning after the first round. If I use `$info.hide()` in the `else:` it works.

Well the idea is that `hide_info()` sets the mouse mode to confined, that's why it's used there.

I think you uncovered a bug in `global.reset()`, try this:
```diff
diff --git a/global.gd b/global.gd
index 07b7e2d..c69c1f9 100644
--- a/global.gd
+++ b/global.gd
@@ -101,7 +101,7 @@ func reset():
 	start_time = OS.get_system_time_secs()
 	time_passed = 0
 	time_paused = 0
-	last_pause_time = 0
+	last_pause_time = start_time
 	game_over = false
 	score = 0
 	combo = 0
diff --git a/level/ui.gd b/level/ui.gd
index 88de755..627c5bd 100644
--- a/level/ui.gd
+++ b/level/ui.gd
@@ -21,11 +21,12 @@ const RANDOM_BITS = [
 
 func _ready():
 	$game_over.hide()
-	$info.hide()
 
 	if global.first_run:
 		global.first_run = false
 		show_info()
+	else:
+		hide_info()
 
 	$game_over/random_bit.set_bbcode(RANDOM_BITS[randi() % RANDOM_BITS.size()])
 
@@ -66,6 +67,7 @@ func _process(_delta):
 
 
 func game_over():
+	Input.set_mouse_mode(Input.MOUSE_MODE_VISIBLE)
 	$AnimationPlayer.play("game_over")
 	yield($AnimationPlayer, "animation_finished")
 	$AnimationPlayer.play("highlight_button")

```

This is in place of changing the mouse mode in `global.restart()`.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**